tools/wltests/test_series

```diff
@@ -812,7 +812,7 @@ name_sha1() {
 			--sort=-committerdate \
 			--format='%(objectname:short) %(refname:short)' \
 			refs/heads/ refs/remotes/ refs/tags |
-	grep $COMMIT_SHA1 | awk '{print $2}' > tmp_pipe &
+	grep ${COMMIT_SHA1:0:12} | awk '{print $2}' > tmp_pipe &
```

Comment:
wouldn't it be better to just use `%(refname)` in the for-each-ref command ? According to the doc, `%(refname:short)` does not give 12 chars, it gives something unambiguous which may be smaller or bigger than that.

Reply:
Mmm... good point, I suppose you mean for `objectname`, that's what generates the SHA1.

If I remove `:short` from objectname I get a non-ambiguous SHA1, which you right it can be longher then 12 chars. The main issues is that in a series file we don't want to use very long SHA1 either and thus, in general, we use 12 char strings which is still potentially a source of ambiguity.

Let's fix at least one ambiguity, I've updated the patch which a prefix check on the full SHA1 returned by the `for-each-ref` query...

Reply:
Rather than prefix check, you can use `COMMIT_SHA1=$(git rev-parse --quiet --verify "$COMMIT_SHA1")` to convert the short format used in the configuration into a full-length sha1. It then becomes comparable with the full sha1 given by `%(objectname)`. If the sha1 is ambiguous, `git rev-parse` will exit with non-zero code and give an empty string so it's quite easy to catch.

Reply:
Also I'm not 100% sure of what the code is doing but it seems like `git name-rev --name-only $COMMIT_SHA1` could replace this `for-each-ref` + `grep` + `awk`.

Reply:
Right... the branch name resolution code is quite convoluted right now.
Let's fix it and add in that check for unique and full SHA1s...